autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -1,6 +1,133 @@
+    meeting_object.save()
+
+def parse_instructor(course_data):
+    """called from parse course. puts instructor data in database"""
+    #gets values to create object
+    for i in course_data['faculty']:
+        instructor_id = i['displayName']#primary key. name is currently being used
+        instructor_email = i['emailAddress']
+        #creates and saves instructor object
+        instructor_object = Instructor(
+            id=instructor_id, email_address=instructor_email)
+        instructor_object.save()
+        return instructor_object
+
+def parse_course(course_data):
+    """puts course data in database. calls parse instructor and section"""
+    course_dept = course_data['subject']
+    course_number = course_data['courseNumber']
+    course_title = course_data['courseTitle']
+    course_term_code = course_data['term']
+    course_id = generate_course_id(
+        course_dept, course_number, course_term_code)#primary key
+    course_credit_hours = course_data['creditHourLow']
+    #creates and saves course object
+    course_object = Course(
+        id=course_id, dept=course_dept, course_num=course_number, title=course_title,
+        credit_hours=course_credit_hours, term=course_term_code)
+    course_object.save()
+    #calls other parse fuctions
+    instructor_object = parse_instructor(course_data)
+    parse_section(course_data, instructor_object)
+    print(f'{course_data}\n')
+
+def get_department_names(banner: BannerRequests) -> List[str]:
+    depts = banner.get_departments()
+    return [dept["code"] for dept in depts]
 
 class Command(base.BaseCommand):
-    "Iterates through all of the courses and loads them into the database"
-    def handle(self, *args, **kwargs):
-        pass
\ No newline at end of file
+    " Gets course information from banner and adds it to the database "
+
+    def add_arguments(self, parser):
+        parser.add_argument('--term', type=str, required=False, default="201931")
+
+    def handle(self, *args, **options):
+        banner = BannerRequests(options['term'])
+        loop = asyncio.get_event_loop()
+        # the next line should be used in production, but we should only use two or three
+        # departments when testing to make the code run way faster
+        # once department scraping is done, we can replace get_department_names
+        # with just getting each of the departments for the term from our database
+        # depts = get_department_names(banner)
+        depts = ['CSCE'] # test with one department, change this if you want
+        json = loop.run_until_complete(banner.search(depts, 10)) # only get a few courses
+        #start = time.time()
+        for course_list in json:
+            for course in course_list:
+                parse_course(course)
+        #finish = time.time()
+        #elapsed_time = finish - start
+        #print(f'Finished scraping in {elapsed_time:.2f} seconds')
```

Comment:
I removed them with the print statement because I assumed the use was for debugging and testing stuff.

Reply:
Nah, I like to have it for comparing speeds just in general, as the end-result for scrape courses (and really scraping in general) should have it so that it scrapes _all_ of the courses going back a few years (needed for grade distributions), which will end up taking a few minutes.